### PR TITLE
form converts comments to entities and stores them in datastore

### PR DESCRIPTION
**Pull request by smruthibalajee**
**Description:** Edited the doPost function to utilize Datastore for storing comments instead of an ArrayList (which isn't a permanent way of storing data). Now, creates entities from the input comments and adds them to the data store. Next step will be to load the data onto the webpage upon user request. Also, edited the randomization of the welcome button messages so that no two messages are displayed in a row.

Screenshot 2020-06-04 at 12 25 14 PM

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -42,10 +41,20 @@ function writeText(inputText, container) {
     typeWriter();
 }
 
-/** Function that calls writeText for the welcome messages. */
+/** Function that calls writeText for the welcome messages. Ensures that no greeting is repeated twice in a row.*/
+var oldIndex = 0;
```
Comment (sabadas) [inline portfolio/src/main/webapp/script.js:45]: might want to highlight this change in your PR description

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -45,6 +48,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    // Define a data store object.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (sabadas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: General question, is there an advantage of creating the datastoreservice on every doPost, or can you just create it once and re-use the same object?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -54,27 +61,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+      populateDataStore(datastore, name, "other", msg);
     }
 
     //redirect back to original page
     response.sendRedirect("/index.html#comment");
   }
 
-  /** Helper function to add the name, category, and message to the right ArrayList. */
+  /** Helper function to add the name, category, and message to the ArrayList. */
   private void populateArrayList(List c, String name, String type, String msg) {
```
Comment (sabadas) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: if this method is not being called anymore, you can delete it

**Review by sabadas — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (smruthibalajee) [reply portfolio/src/main/webapp/script.js:45]: Oh yeah, I completely forgot that I did this yesterday! Just updated!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (smruthibalajee) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: fixed!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (smruthibalajee) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: In the example they provided, the object was created in the doPost function, so I followed that format. I just moved it outside the function to the class!

**Pull request merged**